data-plane/config/broker/500-dispatcher.yaml

```diff
@@ -134,18 +130,21 @@ spec:
 
           livenessProbe:
             failureThreshold: 3
-            tcpSocket:
-              port: 9090
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                - "ls /proc/1"
             initialDelaySeconds: 10
             periodSeconds: 3
             successThreshold: 1
             timeoutSeconds: 1
           readinessProbe:
             failureThreshold: 3
-            httpGet:
-              port: 9090
-              path: /metrics
-              scheme: HTTP
+            exec:
+              command:
+                - cat
+                - /tmp/healthy
```

Comment:
@matzew @creydr I had to switch the liveness/readiness probes because we can no longer be sure that prometheus is running in the data plane.

This means for the dispatcher there is no running http server for the probes to go to, hence the "exec" probes I have here

Reply:
This is something I don't understand. Why can't we be sure the metrics are exposed?

Reply:
The issue is that the user can choose to use either prometheus or otlp for the metrics. If they use otlp, this is a "push" protocol so there is no web server that starts and listens on any port